apix-integrationtests/alfresco/src/main/java/eu/xenit/apix/rest/v1/tests/CommentsTest.java

```diff
@@ -44,21 +44,21 @@ public class CommentsTest extends RestV1BaseTest {
     private static final String SKIPCOUNT = "skipcount";
     private static final String PAGESIZE = "pagesize";
 
-    @Autowired
     private CommentService alfrescoCommentService;
-
-    @Autowired
     private ICommentService commentService;
-
-    @Autowired
-    private ApixToAlfrescoConversion apixConverter;
-
-    @Autowired
     private NodeService nodeService;
 
-    @Autowired
-    private RetryingTransactionHelper transactionHelper;
+    public CommentsTest(){
+        // initialise the local beans
+        nodeService = (eu.xenit.apix.alfresco.metadata.NodeService) testApplicationContext.getBean(eu.xenit.apix.alfresco.metadata.NodeService.class);
```

Comment:
should we try to abstract the `getBean` functionality to the base class and just ask for the type?
Something like

base class:
```
    public <T> T getBean(Class<T> beanClass) {
        return testApplicationContext.getBean(beanClass);
    }
```

Invocation:
`nodeService = getBean(eu.xenit.apix.alfresco.metadata.NodeService.class);`

Reply:
After trying to update the Thread ApplicationContext using `extends AbstractRefreshableApplicationContext, implements ApplicationContextAware`. It is clear that this won't be possible in the current setup since the IntegrationTest code does not trigger the application context initialization process.